File: Backend/Infrastructure.MyShop/Models/DTO/CategoryDTO/CategoryItemDTO.cs
using Infrastructure.MyShop.Models.DTO.ProductDTO;

namespace Infrastructure.MyShop.Models.DTO.CategoryDTO
{
    public class CategoryItemDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Image { get; set; }
        public string Description { get; set; }
        public string DateCreated { get; set; }
        public string DateUpdated { get; set; }
        public int? ParentId { get; set; }
        public string ParentName { get; set; }
        public int countProducts { get; set; }
        public List<CategoryItemDTO> Subcategories { get; set; }
        public List<ProductItemDTO> Products { get; set; }
    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/dto/categoryDTO/CategoryItemDTO.java
package org.example.Infrastructure.dto.categoryDTO;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.example.Infrastructure.dto.productDTO.ProductItemDTO;

import java.io.Serializable;
import java.util.List;

@Data
public class CategoryItemDTO implements Serializable {
    private int id;
    private String name;
    private String image;
    private String description;
    private String dateCreated;
    private String dateUpdated;
    private Integer parentId;
    private String parentName;
    private Integer countSubategories;
    private Integer countProducts;
    @JsonProperty("Subcategories")
    private List<CategoryItemDTO> subcategories;
    @JsonProperty("Products")
    private List<ProductItemDTO> products;
}
